pkg/config/config_test.go

```diff
@@ -274,36 +274,59 @@ func TestLoad(t *testing.T) {
 			name:             "global file does not exist and hosts file exist",
 			globalConfigPath: "",
 			hostsConfigPath:  hostsFilePath,
-			wantGitProtocol:  "https",
+			wantGitProtocol:  "",
 			wantToken:        "yyyyyyyyyyyyyyyyyyyy",
 		},
 		{
 			name:             "global file exist and hosts file does not exist",
 			globalConfigPath: globalFilePath,
 			hostsConfigPath:  "",
 			wantGitProtocol:  "ssh",
-			wantGetErr:       true,
+			wantToken:        "",
+		},
+		{
+			name:             "global file does not exist and hosts file does not exist with no fallback",
+			globalConfigPath: "",
+			hostsConfigPath:  "",
+			wantGitProtocol:  "",
+			wantToken:        "",
+		},
+		{
+			name:             "global file does not exist and hosts file does not exist with fallback",
+			globalConfigPath: "",
+			hostsConfigPath:  "",
+			fallback:         ReadFromString(testFullConfig()),
+			wantGitProtocol:  "ssh",
+			wantToken:        "yyyyyyyyyyyyyyyyyyyy",
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			cfg, err := load(tt.globalConfigPath, tt.hostsConfigPath)
+			cfg, err := load(tt.globalConfigPath, tt.hostsConfigPath, tt.fallback)
 			if tt.wantErr {
 				assert.EqualError(t, err, tt.wantErrMsg)
 				return
 			}
 			assert.NoError(t, err)
-			protocol, err := cfg.Get([]string{"git_protocol"})
-			assert.NoError(t, err)
-			assert.Equal(t, tt.wantGitProtocol, protocol)
-			token, err := cfg.Get([]string{"hosts", "enterprise.com", "oauth_token"})
-			if tt.wantGetErr {
-				assert.EqualError(t, err, `could not find key "hosts"`)
+
+			if tt.wantGitProtocol == "" {
+				assertNoKey(t, cfg, []string{"git_protocol"})
 			} else {
-				assert.NoError(t, err)
+				assertKeyWithValue(t, cfg, []string{"git_protocol"}, tt.wantGitProtocol)
+			}
+
+			if tt.wantToken == "" {
+				assertNoKey(t, cfg, []string{"hosts", "enterprise.com", "oauth_token"})
+			} else {
+				assertKeyWithValue(t, cfg, []string{"hosts", "enterprise.com", "oauth_token"}, tt.wantToken)
+			}
+
+			if tt.fallback != nil {
+				// Assert that load returns an equivalent copy of fallvback.
+				assert.Equal(t, tt.fallback.entries.String(), cfg.entries.String())
```

Comment:
Man, I totally get why the yaml Node has a slice for content now rather than a map, it's so it can always marshal in a stable order!